Add Message Content Intent section (#5394)
hammerandchisel/discord-api-docs: docs/topics/Gateway.md

### Privileged Intents

Some intents are defined as "Privileged" due to the sensitive nature of the data. Currently, those intents include:

- `GUILD_PRESENCES`
- `GUILD_MEMBERS`
- [`MESSAGE_CONTENT`](#DOCS_TOPICS_GATEWAY/message-content-intent)

> Apps that qualify for verification must be approved for the privileged intent(s) [during verification](https://support.discord.com/hc/en-us/articles/360040720412-Bot-Verification-and-Data-Whitelisting) before they can use them. If your app is already verified and you need to request additional privileged intents, [contact support](https://dis.gd/support).

Reviewer: Intents must be requested through the dev portal, not through support.

```suggestion
Apps that qualify for verification must be approved for the privileged intent(s) [during verification](https://support.discord.com/hc/en-us/articles/360040720412-Bot-Verification-and-Data-Whitelisting) before they can use them. If your app is already verified, you can request additional privileged intents in the Developer Portal.
```

Also I think the verification team now wants intents to be requested after verification instead of during verification; apps that request intents during verification are often asked to re-apply for intents after verification. that should be clarified and maybe updated here

> To specify privileged intents in your `IDENTIFY` payload, you must enable the priviledged intents your app needs. To toggle priviledged intents, navigate to your app's settings in the Developer Portal. Click on the **Bot** page, and under the "Privileged Gateway Intents" section you can enable the toggle for each intent your app needs. Verified apps can only use privileged intents once they've been approved for them.

Reviewer: typo
```suggestion
To specify privileged intents in your `IDENTIFY` payload, you must enable the privileged intents your app needs. To toggle privileged intents, navigate to your app's settings in the Developer Portal. Click on the **Bot** page, and under the "Privileged Gateway Intents" section you can enable the toggle for each intent your app needs. Verified apps can only use privileged intents once they've been approved for them.
```

Author: it's impressive how consistently i spell this wrong

> info
> Unverified apps can use privileged intents without approval, but still must enable them in their app's settings. If the app's verification status changes, it will then have to apply for the privileged intent(s).

Events under the `GUILD_PRESENCES` and `GUILD_MEMBERS` intents are turned **off by default on all API versions**. If you are using **API v6**, you will receive those events if you are authorized to receive them and have enabled the intents in the Developer Portal. You do not need to use Intents on API v6 to receive these events; you just need to enable the flags. If you are using **API v8** or above, Intents are mandatory and must be specified when identifying.

In addition to the gateway restrictions described here, Discord's REST API is also affected by Privileged Intents. For example, to use the [List Guild Members](#DOCS_RESOURCES_GUILD/list-guild-members) endpoint, you must have the `GUILD_MEMBERS` intent enabled for your application. This behavior is independent of whether the intent is set during `IDENTIFY`.

#### Message Content Intent

> `MESSAGE_CONTENT (1 << 15)` is a unique privileged intent that isn't directly associated with any Gateway events. Instead, access to `MESSAGE_CONTENT` permits your app to receive message content data across the APIs, including (but not limited to) the `content`, `embeds`, `attachments`, and `components` fields.

Reviewer: > including (but not limited to) the `content`, `embeds`, `attachments`, and `components` fields

according to everything else this is an exhaustive list; if it isn't then the full list should be published, instead of a partial list.

```suggestion
`MESSAGE_CONTENT (1 << 15)` is a unique privileged intent that isn't directly associated with any Gateway events. Instead, access to `MESSAGE_CONTENT` permits your app to receive message content data across the APIs, specifically the `content`, `embeds`, `attachments`, and `components` fields.
```

Author: there's `matched_content` as well: https://discord.com/developers/docs/topics/gateway#auto-moderation-action-execution-auto-moderation-action-execution-event-fields

i'm not sure we just want to be listing out random fields as an exhaustive list here—they probably make more sense in the references for the specific responses.

Reviewer: maybe something like this?
```suggestion
`MESSAGE_CONTENT (1 << 15)` is a unique privileged intent that isn't directly associated with any Gateway events. Instead, access to `MESSAGE_CONTENT` permits your app to receive message content data across the APIs. For the message object, this applies to the `content`, `embeds`, `attachments`, and `components` fields; for other objects, the affected fields (if any) are listed in that object's documentation.
```

Author: Updated this a bit—lmk if confusing. I can take a deeper pass after I fold this into the updated Gateway documentation if it still needs some work

> info

> > Like other privileged intents, `MESSAGED_CONTENT` must be approved during verification or afterwards by contacting support. You can read more about the message content intent review policy [in the Help Center](https://support-dev.discord.com/hc/en-us/articles/5324827539479).

Reviewer: ```suggestion
> Like other privileged intents, `MESSAGE_CONTENT` must be approved during verification or afterwards by applying in the Developer Portal. You can read more about the message content intent review policy [in the Help Center](https://support-dev.discord.com/hc/en-us/articles/5324827539479).
```
